src/server/index.ts

```diff
@@ -9,28 +9,56 @@ import bodyParser from 'body-parser';
 import { configureStore } from '../shared/store';
 import serverRender from './render';
 import paths from '../../config/paths';
+import { ErrorRequestHandler, Handler, RequestHandler, Request } from 'express-serve-static-core';
+import { Store } from 'redux';
 
 require('dotenv').config();
 
 const app = express();
 
+interface RequestWithStore extends Request {
+    store?: Store;
+}
+
+const errorHandler: ErrorRequestHandler = (err, _, res) =>
+    res.status(404).json({
+        status: 'error',
+        message: err.message,
+        stack:
+            // print a nicer stack trace by splitting line breaks and making them array items
+            process.env.NODE_ENV === 'development' &&
+            (err.stack || '')
+                .split('\n')
+                .map((line: string) => line.trim())
+                .map((line: string) => line.split(path.sep).join('/'))
+                .map((line: string) =>
+                    line.replace(
+                        process
+                            .cwd()
+                            .split(path.sep)
+                            .join('/'),
+                        '.'
+                    )
+                ),
+    });
+
 // Use Nginx or Apache to serve static assets in production or remove the if() around the following
 // lines to use the express.static middleware to serve assets for production (not recommended!)
 if (process.env.NODE_ENV === 'development') {
     app.use(paths.publicPath, express.static(path.join(paths.clientBuild, paths.publicPath)));
-    app.use('/favicon.ico', (req: express.Request, res: express.Response) => {
-        res.send('');
-    });
+    app.use('/favicon.ico', (_, res) => res.send(''));
```

Comment:
If find it good custom to give unused variables the names `_` or `__`, etc. Makes it clear that you will not use them.

Reply:
Personally I'm not a big fan of that pattern. I know it's somewhat common to use `_` or `__` to indicate it's an unused parameter but I prefer to disable it using the `eslint-disable` comment for two reasons:
- it's more explicit, especially for unexperienced developers who haven't heard of that `_` "mechanism" to mark unused parameters (I remember how confused I was when I first came across such a variable in a mostly unknown codebase)
- you don't have to rename your variable if you later decide to do use them (kind of esotheric, I know)

Reply:
You can use whatever name. If you have an "unused" parameter but a "used" parameter as the last one, tslint will not give an error. Because how else could you write that code? The tslint error only comes when the last parameter(s) are unused.